libs/ledger-live-common/src/families/cosmos/js-buildTransaction.ts

```diff
@@ -43,193 +65,239 @@ export const buildTransaction = async (
                 amount: transaction.amount.toString(),
               },
             ],
-          },
+          }).finish(),
         });
       }
       break;
-
     case "delegate":
-      if (!transaction.validators || transaction.validators.length < 1) {
-        isComplete = false;
-      } else {
+      if (transaction.validators && transaction.validators.length > 0) {
         const validator = transaction.validators[0];
-        if (!validator) {
-          isComplete = false;
-          break;
-        } else if (!validator.address || transaction.amount.lte(0)) {
-          isComplete = false;
-        }
-
-        msg.push({
-          typeUrl: "/cosmos.staking.v1beta1.MsgDelegate",
-          value: {
-            delegatorAddress: account.freshAddress,
-            validatorAddress: validator.address,
-            amount: {
-              denom: account.currency.units[1].code,
-              amount: transaction.amount.toString(),
+        if (validator && validator.address && transaction.amount.gt(0)) {
+          //todo this should not be gt0, but the minimum required by osmosis to stake
```

Comment:
Should remove this osmosis-specific comment. Any way to fix it too?

Reply:
no fix needed. there is no minimum required by osmosis